rstest_macros/src/parse/rstest/files.rs

```diff
@@ -337,6 +368,20 @@ impl ValueFilesExtractor {
             },
         )
     }
+
+    fn extract_file_mode(&mut self, node: &mut FnArg) -> Vec<LitStrAttr> {
+        self.extract_argument_attrs(
+            node,
+            |a| attr_is(a, "file_mode"),
+            |attr| {
+                LitStrAttr::parse_meta_name_value(&attr).map_err(|_| {
+                    attr.error(
+                        "Use #[file_mode = \"...\"] to define the argument of the file input",
+                    )
+                })
```

Comment:
Here you can try to parse directly a `FilesMode` (should implement a `Parse` trait) that get an `Ident` and try to match it whit the expected varaints.